### PR TITLE
remove star imports

### PR DESCRIPTION
**Pull request by simbilod**
star imports clutter the namespace and make it difficult for automated tools to navigate the codebase. See [official Python documentation](https://docs.python.org/3/faq/programming.html#what-are-the-best-practices-for-using-import-in-a-module) and [PEP8 guide](https://peps.python.org/pep-0008/#imports).

I refactored all Python module/example/testing files to explicitly import functions instead of using star imports whenever possible. 

I have run all the files under testing/ and examples/, and they all execute without import errors.

In general this is a first step toward enforcing style guides to improve readability.

### CODE REVIEW

**Comment by tcaduser** [comment]
great idea.  did you do this all by hand?   this is a good idea.  for example, the `devsim` exception is just called `devsim.error`.  I'll review and run with the regression tests

File: testing/GaussFermi_float128.py [hunk 1]
```diff
@@ -16,5 +15,3 @@
 devsim.set_parameter(name = "extended_model", value=True)
 devsim.set_parameter(name = "extended_equation", value=True)
 
-import GaussFermi
```
Comment (tcaduser) [inline testing/GaussFermi_float128.py:19]: This import loads and runs `GaussFermi.py` as a script after setting extended precision.

File: testing/testfunc_extended.py [hunk 2]
```diff
@@ -4,4 +4,3 @@
 ###
 import devsim
 devsim.set_parameter(name="extended_model", value=True)
-import testfunc
```
Comment (tcaduser) [inline testing/testfunc_extended.py:7]: needed to  run the test as a script

File: testing/kahan_float128.py [hunk 3]
```diff
@@ -16,5 +15,3 @@
 devsim.set_parameter(name = "extended_model", value=True)
 devsim.set_parameter(name = "extended_equation", value=True)
 
-import kahan
```
Comment (tcaduser) [inline testing/kahan_float128.py:19]: this is needed to run the test

File: testing/Fermi1_float128.py [hunk 4]
```diff
@@ -15,5 +15,3 @@
 devsim.set_parameter(name = "extended_model", value=True)
 devsim.set_parameter(name = "extended_equation", value=True)
 
-import Fermi1
```
Comment (tcaduser) [inline testing/Fermi1_float128.py:18]: needed to run test

File: testing/mos_2d_params.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import devsim
```
Comment (tcaduser) [inline testing/mos_2d_params.py:1]: This file is generated in another test.

Comment (simbilod) [reply testing/mos_2d_params.py]: fixed!

File: testing/Fermi1_float128.py (hunk 4 above)
Comment (simbilod) [reply testing/Fermi1_float128.py:18]: fixed!

File: testing/kahan_float128.py (hunk 3 above)
Comment (simbilod) [reply testing/kahan_float128.py:19]: fixed!

File: testing/testfunc_extended.py (hunk 2 above)
Comment (simbilod) [reply testing/testfunc_extended.py:7]: fixed!

File: testing/GaussFermi_float128.py (hunk 1 above)
Comment (simbilod) [reply testing/GaussFermi_float128.py:19]: fixed!

**Comment by simbilod** [comment]
> great idea. did you do this all by hand? this is a good idea. for example, the `devsim` exception is just called `devsim.error`. I'll review and run with the regression tests

I used a mix of search/replace, code inspection, and Python packages to remove unused imports. 

I wasn't sure how to automatically run the tests and just looped over the files with a bash script, thanks for catching what I missed!

**Comment by tcaduser** [comment]
thank you.  The tests are run via ctest and the CMakeLists.txt file in the testing directory.  It requires the related repos containing test results.  it gets complicated since it requires an exact numerical match.  I can provide more information in the future.

**Pull request merged**